xgi/generators/classic.py

```diff
@@ -154,3 +157,70 @@ def trivial_hypergraph(n=1, create_using=None, default=None):
     H.add_nodes_from(nodes)
 
     return H
+
+
+def complete_hypergraph(N, order=None, max_order=None, include_singletons=False):
+    """
+    Generate a complete hypergraph, i.e. one that contains all possible hyperdges
+    at a given `order` or up to a `max_order`.
+
+    Parameters
+    ----------
+
+    N : int
+        Number of nodes
+    order : int or None
+        If not None (default), specifies the single order for which to generate hyperedges
+    max_order : int or None
+        If not None (default), specifies the maximum order for which to generate hyperedges
+    include_singletons : bool
+        Whether to include singleton edges (default: False). This argument is discarded
+        if max_order is None.
+
+    Return
+    ------
+    Hypergraph object
+        A complete hypergraph with `N` nodes
+
+    Notes
+    ----
+    Only one of `order` and `max_order` can be specified by and int (not None).
+    Additionally, at least one of either must be specified.
+
+    The number of possible edges grows exponentially as :math:`2^N` for large `N` and
+    quickly becomes impractically long to compute, especially when using `max_order`. For
+    example, `N=100` and `max_order=5` already yields :math:`10^8` edges. Increasing `N=1000`
+    makes it :math:`10^{13}`. `N=100` and with a larger `max_order=6` yields :math:`10^9` edges.
+
+
+    """
+    # this import needs to happen when the function runs, not when the module is first
+    # imported, to avoid circular imports
+    import xgi
+
+    if (order is None) and (max_order is None):
+        raise ValueError(
+            "At least one among order and max_order must be specified (not None)"
+        )
+    if (order is not None) and (max_order is not None):
+        raise ValueError(
+            "Both order and max_order cannot be specified (not None) at the same time."
+        )
+
+    H = xgi.Hypergraph()
+
+    nodes = range(N)
+    H.add_nodes_from(nodes)
+
+    if order is not None:
+        edges = combinations(nodes, order + 1)
+    elif max_order is not None:
+        start = 1 if include_singletons else 2
+        end = max_order + 1
```

Comment:
Maybe add `assert end > start`

Reply:
Done. Can also be = because adding +1 to end in the range two lines down.